### DISCUSSION TITLE
dot't allow use "blob" field,how i can?

### DISCUSSION THREAD

**Discussion by SummerZYQ** [quartznet/quartznet#1200] [Q&A] [answered]
i am use Quartz.AspNetCore3.3.2 in asp.net core 3.1
use mysql and cluster
table use this sql : quartznet/database/tables/tables_mysql_innodb.sql
but we dba not allow to use "blob" and "text" field
how i can simply to modify "blob" to "varchar"

**Comment by lahma** [comment, marked as answer]
Unfortunately you probably need to write some custom code to handle all the inserts and updates (IDriverDelegate, maybe inheriting from the MySQL one) and send a thank you bill to DBA which determined that this was somehow reasonable course of action.

**Reply by SummerZYQ** [reply]
thank, I will try inheriting the MySQLDelegate and override all method that has the blob field.
I use json serializer, so, I only copy base method and  convert byte[] to string. that is ok?

**Reply by lahma** [reply]
Probably, let me know if there's more variation points required.

**Reply by SummerZYQ** [reply]
amazing, i am find a so easy way to get it.
first,i am sure use JsonSerializer.
override ReadBytesFromBlob and update it to get string and return Encoding.UTF8.GetBytes.
override AddCommandParameter, if dateType is binary, update value from Encoding.UTF8.GetString and write it.
sorry,i am a Junior developer and english is bad, can you help me to ensure this way can use to production?

**Reply by lahma** [reply]
Sounds that that might work. Just write enough integration tests so that you know it will be covered and won't break later on with some library update.

**Reply by SummerZYQ** [reply]
thanks for your reply. i try to do it.
